avm-transpiler/src/transpile.rs

```diff
@@ -330,7 +374,12 @@ pub fn brillig_to_avm(brillig_bytecode: &[BrilligOpcode<FieldElement>]) -> (Vec<
                 handle_foreign_call(&mut avm_instrs, function, destinations, inputs);
             }
             BrilligOpcode::BlackBox(operation) => {
-                handle_black_box_function(&mut avm_instrs, operation);
+                handle_black_box_function(
+                    &mut avm_instrs,
+                    operation,
+                    &mut procedures_used,
+                    &mut unresolved_jumps,
```

Comment:
boo, this really needs to go into an object so that all the functions can access them :) but not your fault! no need to do it in this PR